core/src/test/java/org/apache/calcite/test/SqlFunctionsTest.java

```diff
@@ -810,6 +826,674 @@ private void thereAndBack(byte[] bytes) {
+    }
+
+    // wrapper behavior test
+
+    assertThat(
+        SqlFunctions
+            .jsonQuery(
+                SqlFunctions.PathContext
+                    .withReturned(SqlFunctions.PathMode.STRICT, "bar"),
+                SqlJsonQueryWrapperBehavior.WITH_UNCONDITIONAL_ARRAY,
+                SqlJsonQueryEmptyOrErrorBehavior.NULL,
+                SqlJsonQueryEmptyOrErrorBehavior.NULL),
+        is("[\"bar\"]")
+    );
+
+    assertThat(
+        SqlFunctions
+            .jsonQuery(
+                SqlFunctions.PathContext
+                    .withReturned(SqlFunctions.PathMode.STRICT, "bar"),
+                SqlJsonQueryWrapperBehavior.WITH_CONDITIONAL_ARRAY,
+                SqlJsonQueryEmptyOrErrorBehavior.NULL,
+                SqlJsonQueryEmptyOrErrorBehavior.NULL),
+        is("[\"bar\"]")
+    );
+
+    assertThat(
+        SqlFunctions
+            .jsonQuery(
+                SqlFunctions.PathContext
+                    .withReturned(SqlFunctions.PathMode.STRICT, Collections.singletonList("bar")),
+                SqlJsonQueryWrapperBehavior.WITH_UNCONDITIONAL_ARRAY,
+                SqlJsonQueryEmptyOrErrorBehavior.NULL,
+                SqlJsonQueryEmptyOrErrorBehavior.NULL),
+        is("[[\"bar\"]]")
+    );
+
+    assertThat(
+        SqlFunctions
+            .jsonQuery(
+                SqlFunctions.PathContext
+                    .withReturned(SqlFunctions.PathMode.STRICT, Collections.singletonList("bar")),
+                SqlJsonQueryWrapperBehavior.WITH_CONDITIONAL_ARRAY,
+                SqlJsonQueryEmptyOrErrorBehavior.NULL,
+                SqlJsonQueryEmptyOrErrorBehavior.NULL),
+        is("[\"bar\"]")
+    );
+  }
+
+  @Test public void testJsonize() {
+    assertThat(
+        SqlFunctions.jsonize(new HashMap<>()),
+        is("{}")
+    );
+  }
+
+  @Test public void testDejsonize() {
+    assertThat(
+        SqlFunctions.dejsonize("{}"),
+        is(Collections.emptyMap())
+    );
+    assertThat(
+        SqlFunctions.dejsonize("[]"),
+        is(Collections.emptyList())
+    );
+
+    // expect exception thrown
+    try {
+      SqlFunctions.dejsonize("[}");
+      fail("expect exception, but not");
+    } catch (Exception ignored) {
+      // ignored
+    }
+  }
+
+  @Test public void testJsonObject() {
+    assertThat(
+        SqlFunctions.jsonObject(SqlJsonConstructorNullClause.NULL_ON_NULL),
+        is("{}")
+    );
+    assertThat(
+        SqlFunctions.jsonObject(SqlJsonConstructorNullClause.NULL_ON_NULL,
+            "foo",
+            "bar"),
+        is("{\"foo\":\"bar\"}")
+    );
+    assertThat(
+        SqlFunctions.jsonObject(SqlJsonConstructorNullClause.NULL_ON_NULL,
+            "foo",
+            null),
+        is("{\"foo\":null}")
+    );
+    assertThat(
+        SqlFunctions.jsonObject(SqlJsonConstructorNullClause.ABSENT_ON_NULL,
+            "foo",
+            null),
+        is("{}")
+    );
+  }
+
+  @Test public void testJsonObjectAggAdd() {
```

Comment:
Would you please ensure the tests print human-understandable messages in case of failures?